engine/core/src/main/java/org/datacleaner/util/LabelUtils.java

```diff
@@ -96,13 +97,26 @@ public static String getLabel(ComponentJob job) {
      * 
      * @return
      */
-    public static String getLabel(ComponentJob job, boolean includeDescriptorName, boolean includeInputColumnNames,
+    public static String getLabel(ComponentJob job, boolean includeDescriptorName, boolean includeInputColumnNames, boolean includeRequirements) {
+        return getLabel(job, null, includeDescriptorName, includeInputColumnNames, includeRequirements);
+    }
+
+    public static String getLabel(ComponentJob job, AnalyzerResult result, boolean includeDescriptorName, boolean includeInputColumnNames,
             boolean includeRequirements) {
         final String jobName = job.getName();
         final StringBuilder label = new StringBuilder();
         if (Strings.isNullOrEmpty(jobName)) {
             ComponentDescriptor<?> descriptor = job.getDescriptor();
-            label.append(descriptor.getDisplayName());
+            String displayName = descriptor.getDisplayName();
+            if (result != null) {
+                if (result instanceof HasLabelAdvice) {
```

Comment:
I can see your point that it's difficult to invoke any method on the component object from where you are in the code here. But ideally speaking the DC framework should take the burden away from the extension developer and I guess also it would be preferable if the label/name of a component is represented in the same way on the desktop UI graph as well as in the result screen. Currently the HasLabelAdvice does influence the label/name on the graph - just not the result screen. So having a (potentially different) label on the result screen could end up feeling a bit chaotic/inconsistent.

The ComponentJob is what DC uses (mostly internally although it has a public interface) to represent the component and it's configuration without actually instantiating it. At the same time the component itself and HasLabelAdvice is implemented by the extension developer, which probably doesn't care about the ComponentJob representation.

One aspect of ComponentJob is that it has a getName() method. We could consider if we wanted to apply a name to the ComponentJob at the time of building it (at that time there _is_ a component instance available). That way the LabelUtils class can stay as it was because it is already checking if there is a name. If we can go down this route to solving the issue, then you should be looking into ComponentBuilder and it's implementing classes.